tests-extensive/test_extreme_functional.py

```
@@ -405,34 +379,120 @@ def _capture_diagnostics():
-        pid = _wait_for_player(timeout=600)
-    except Exception:  # noqa: BLE001
-        # Connection reset / refused while polling = Kodi crashed mid-test.
-        # Snapshot diagnostics first so the cause is reachable post-teardown.
+    def _clear_kodi_dialogs():
+        """Dismiss any leftover modal (e.g. the addon's 'Download failed'
+        dialog) so a retry's ExecuteAddon isn't swallowed by it."""
+        for _ in range(3):
+            try:
+                _kodi_rpc("Input.Select")
+            except Exception:  # noqa: BLE001
+                pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)